agatha/construct/semrep_util.py

```diff
@@ -302,15 +318,101 @@ def _get_flags(self, input_path:Path, output_path:Path)->List[str]:
     if self.word_sense_disambiguation:
       res.append("--word_sense_disambiguation")
     res.append("--xml_output_format")
-    res.append(str(input_path))
-    res.append(str(output_path))
+    if input_path is not None and output path is not None:
+      res.append(str(input_path))
+      res.append(str(output_path))
+    else:
+      raise ValueError("Both input and output paths must be set or None")
     return res
 
-  def run(self, input_path:Path, output_path:Path)->None:
+  def _read_lines(self, in_file:Path)->List[str]:
+    """
+    Reads in lines of a file in memory.
+
+    Args:
+     in_file: Path to input file
+    Returns:
+      lines in the files
+    """
+    with open(in_file, 'r') as f:
+      lines = []
+      for line in f:
+        lines.append(line.strip())
+    return lines
+
+  def _write_xml(self, data: List[lxml.etree._Element], out_file:Path)->None:
+    """
+    Writes the produced XML to the specified
+    file.
+
+    Args:
+      data: List of xml elements produced by semrep
+      out_file: Path to output xml file
+    """
+    assert out_file.suffix == '.xml', 'The output must be a valid XML file'
```

Comment:
while I think its good practice to write xml files with the `.xml` suffix, do we really want to crash if we some day want to write a `.txt` file?

Reply:
So, as I understood from `Semrep` docs, it will always produce an `xml` output. What would be a case when we actually want to write a `.txt`?

Reply:
so semrep has a number of output modes, we select xml when we use the `-X` flag in our statement preparation function. Its other modes, such as the "full-field output" `-F` is a plaintext mode.

But my point is more so that I don't want to crash the training pipeline in the case that the output file path just doesn't have a name that ends in `.xml`. The `.txt` example might not have been clear. Another common case would be using file paths generated by something like `mktemp` or `agatha.construct.file_util.touch_random_unused_file`. While both tools certainly allow you to specify file extensions, its an easy detail to miss later down the road.

Reply:
tl;dr: I'm not sure its worth the eventual debugging time that crashing on this assert is worth. However, I'm sympathetic to the idea that if we're going to write an xml file, it should end in `.xml`